lib/src/package_meta.dart

```diff
@@ -282,17 +282,26 @@ class _FilePackageMeta extends PubPackageMeta {
   FileContents _changelog;
   Map _pubspec;
   Map _analysisOptions;
+  final List<String> _initializationErrors = [];
 
   _FilePackageMeta(Directory dir) : super(dir) {
     var f = File(path.join(dir.path, 'pubspec.yaml'));
     if (f.existsSync()) {
-      _pubspec = loadYaml(f.readAsStringSync());
+      try {
+        _pubspec = loadYaml(f.readAsStringSync());
+      } catch (e) {
+        _initializationErrors.add('Can not parse ${f.path}: $e');
```

Comment:
It looks like in effect you're using `_pubspec == null` as a sentinel value meaning "pubspec is corrupt".  Would it be safer to instead have a `_pubspecIsCorrupt` boolean?  Then you could do:
```dart
try {
  _pubspec = loadYaml(f.readAsStringSync());
} catch (e) {
  _initializationErrors.add('Can not parse ${f.path}: $e');
  _pubspec = {};
  _pubspecIsCorrupt = true;
}
```
And you wouldn't have to worry about whether accesses to `_pubspec` are properly null checked.

(Then again, I don't know if `_pubspecIsCorrupt` is really needed; see my comment below in `getInvalidReasons`)

Reply:
yes, this is bad.  will fix the style of other bits of code to match the fix.

Reply:
Also, better to read the file separately and then catch `FormatException` or similar. 

https://dart.dev/guides/language/effective-dart/usage#dont-discard-errors-from-catches-without-on-clauses